src/intrinsics/node/buffer.js

```diff
@@ -0,0 +1,135 @@
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import invariant from "../../invariant.js";
+import { Realm } from "../../realm.js";
+import { Value, NumberValue, NativeFunctionValue, ObjectValue, StringValue } from "../../values/index.js";
+import { Set, ToInteger } from "../../methods/index.js";
+
+declare var process: any;
+
+function getBufferFromArg(realm: Realm, value: Value): Uint8Array {
+  invariant(value instanceof ObjectValue && value.$ViewedArrayBuffer);
+  let buffer = value.$ViewedArrayBuffer;
+  invariant(buffer instanceof ObjectValue && buffer.$ArrayBufferData);
+  return buffer.$ArrayBufferData;
+}
+
+export default function (realm: Realm): ObjectValue {
+  let nativeBuffer = process.binding('buffer');
+  let nativeBufferPrototype = (require('buffer'): any).Buffer.prototype;
+
+  let intrinsicName = 'process.binding("buffer")';
+  let obj = new ObjectValue(realm, realm.intrinsics.ObjectPrototype, intrinsicName);
+
+  // Buffer
+
+  let setupBufferJS = new NativeFunctionValue(realm, intrinsicName + ".setupBufferJS", "setupBufferJS", 0, (setupContext, setupArgs) => {
+    invariant(setupArgs.length === 2);
+    invariant(setupArgs[0] instanceof ObjectValue);
+    invariant(setupArgs[1] instanceof ObjectValue);
+    let [proto] = setupArgs;
+
+    let simpleWrapperNames = [
+      "asciiSlice",
+      "base64Slice",
+      "latin1Slice",
+      "hexSlice",
+      "ucs2Slice",
+      "asciiWrite",
+      "base64Write",
+      "latin1Write",
+      "hexWrite",
+      "ucs2Write",
+      "utf8Write",
+    ];
+
+    for (let name of simpleWrapperNames) {
+      let wrapper = new NativeFunctionValue(realm, "Buffer.prototype." + name, name, 0, (context, args) => {
+        throw new Error("TODO: " + name);
+      });
+      Set(realm, proto, name, wrapper, true);
+    }
+
+    // utf8Slice is used to read source code.
+    let utf8Slice = new NativeFunctionValue(realm, "Buffer.prototype.utf8Slice", "utf8Slice", 0, (context, args) => {
+      let self = getBufferFromArg(realm, context);
+      let decodedArgs = args.map((arg, i) => {
+        return ToInteger(realm, arg);
+      });
+      let utf8String = nativeBufferPrototype.utf8Slice.apply(self, decodedArgs);
+      return new StringValue(realm, utf8String);
+    });
+    Set(realm, proto, "utf8Slice", utf8Slice, true);
+
+    // copy has recently moved from the prototype to the instance upstream.
+    let copy = new NativeFunctionValue(realm, "Buffer.prototype.copy", "copy", 0, (context, args) => {
+      let self = getBufferFromArg(realm, context);
+      let decodedArgs = args.map((arg, i) => {
+        if (i === 0) {
+          return getBufferFromArg(realm, arg);
+        } else {
+          return ToInteger(realm, arg);
+        }
+      });
+      let bytesCopied = nativeBufferPrototype.copy.apply(self, decodedArgs);
```

Comment:
If you unpacked args with a pattern, you might get away with calling copy directly.